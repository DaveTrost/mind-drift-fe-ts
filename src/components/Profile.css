.ProfileArea {
  margin: 2em;
  overflow: auto; 
  height: 100%; 

  img {
    width: 100px;
    height: 100px;
    border-radius: 50%;
    object-fit: cover;
    margin-bottom: 2em;
  }

  h2 {
    font-family: Montserrat;
    font-weight: 700;
    text-transform: uppercase;
    margin-bottom: 1em;
    color: var(--rust);
  }
  h3 {
    font-family: Montserrat;
    font-size: 14px;
    font-weight: 400;
    line-height: normal;
    color: var(--rust);
  }
  p {
    font-family: Montserrat;
    font-weight: 200;
    line-height: normal;
    font-size: 14px;
    margin-bottom: 0.5em;
  }
}

.Loading {
  font-family: Montserrat;
  font-size: 16px;
  text-align: center;
  color: var(--rust);
}