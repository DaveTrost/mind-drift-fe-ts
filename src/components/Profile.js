import React, { useState, useEffect } from 'react';
import { useAuth0 } from '../react-auth0-spa';
import NavBar from './NavBar';
import styles from './Profile.css';
import Loading from './Loading';
import Achievements from './Achievements';
import { fetchAchievements } from '../services/achievement';

const Profile = () => {
  const { user } = useAuth0();
  const [achieves, setAchieves] = useState([]);
  const loading = !user;
  useEffect(() => {
    fetchAchievements(user.sub)
      .then(list => setAchieves(list));
    return;
  }, []);
  
  return (
    <>
      <Loading loading={loading} />
      <NavBar />
      {!loading &&
        <section className={styles.ProfileArea}>
          <img src={user.picture} alt="Profile" />
          <h2>Your Profile:</h2>
          <p>Hello, {user.name}</p>
          <div>
            <Achievements achieves={achieves} />
          </div>
        </section>
      }
    </>
  );
};

export default Profile;
