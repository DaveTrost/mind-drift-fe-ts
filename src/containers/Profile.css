.ProfileArea {
  margin: 2em;
  overflow: auto; 
  height: 100%; 
  margin-bottom: 4em;

  img {
    width: 100px;
    height: 100px;
    border-radius: 50%;
    object-fit: cover;
    margin-bottom: 2em;
  }

  h2 {
    font-family: Montserrat;
    font-weight: 400;
    font-size: 1.5em;
    text-transform: uppercase;
    margin-bottom: 1em;
    color: var(--dark-blue);
  }
  h3 {
    font-family: Montserrat;
    font-size: 1.2em;
    font-weight: 400;
    line-height: normal;
    color: var(--rust);
  }
  p {
    font-family: Montserrat;
    font-weight: 200;
    line-height: normal;
    font-size: 1em;
    margin-bottom: 0.5em;
  }
}