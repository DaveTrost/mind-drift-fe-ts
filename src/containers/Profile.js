import React, { useState, useEffect } from 'react';
import Achievements from '../components/Achievements';
import Loading from '../components/Loading';
import NavBar from '../components/NavBar';
import Aggs from '../components/Aggs';
import { fetchAchievements } from '../services/achievement';
import { getAverage, getTotal } from '../services/session';
import { fetchUser } from '../services/user';
import User from '../components/User';
import { useSession } from '../utils/WithSession';

const Profile = () => {
  const { user } = useSession();
  const [achieves, setAchieves] = useState([]);
  const [total, setTotal] = useState({});
  const [average, setAverage] = useState({});
  const [streak, setStreak] = useState(0);
  const loading = !user;
  useEffect(() => {
    fetchAchievements(user.uid)
      .then(list => setAchieves(list));

    fetchUser(user.uid)
      .then(([user]) => setStreak(user.currentStreak));

    getTotal(user.uid)
      .then(([total]) => {
        const rounded = makeTimer(total.totalTime);
        setTotal(rounded);
      });

    getAverage(user.uid)
      .then(([average]) => {
        const rounded = makeTimer(average.averageTime);
        setAverage(rounded);
      });
    return;
  }, []);
  
  return (
    <>
      <Loading loading={loading} />
      <NavBar />
      {!loading &&
        <section>
          <User image={user.photoURL} userName={user.displayName} />
          <div>
            <Aggs average={average} total={total} streak={streak}/>
            <Achievements achieves={achieves} />
          </div>
        </section>
      }
    </>
  );
};

const makeTimer = (num) => {
  const timeObj = {
    time: 0,
    type: 'minutes'
  };
  const hours = Math.ceil((num / 60 / 60) * 10) / 10;
  const minutes = Math.ceil((num / 60) * 10) / 10;
  if(hours < 1) {
    timeObj.time = minutes;
  } else {
    timeObj.time = hours;
    timeObj.type = 'hours';
  }
  return timeObj;
};

export default Profile;
