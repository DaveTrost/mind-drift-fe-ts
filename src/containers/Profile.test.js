import React from 'react';
import { shallow } from 'enzyme';
import Profile from './Profile';

import withSessionMock from '../utils/WithSession';
jest.mock('../utils/WithSession', () => ({}));

import achievementServicesMock from '../services/achievement';
jest.mock('../services/achievement', () => ({}));

import sessionServicesMock from '../services/session';
jest.mock('../services/session', () => ({}));

import userServicesMock from '../services/user';
jest.mock('../services/user', () => ({}));

describe('Profile component', () => {
  withSessionMock.useSession = () => ({
    user: {
      displayName: 'me',
      photoURL: 'image_url',
    }
  });

  achievementServicesMock.fetchAchievements = () => ([]);

  sessionServicesMock.getAverage = () => [60];
  sessionServicesMock.getTotal = () => [12345];

  userServicesMock.fetchUser = () => [{ currentStreak: 5 }];

  it('renders the profile', () => {
    const wrapper = shallow(<Profile />);
    expect(wrapper).toMatchSnapshot();
  });
});
