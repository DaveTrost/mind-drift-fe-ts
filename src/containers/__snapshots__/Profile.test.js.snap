// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders the profile 1`] = `
<Fragment>
  <Loading
    loading={false}
  />
  <NavBar />
  <section
    className="ProfileArea"
  >
    <img
      alt="Profile"
      src="http://domain.com/pictures/0000.png"
    />
    <h2>
      Hello, 
      me
    </h2>
    <div>
      <Achievements
        achieves={Array []}
      />
    </div>
  </section>
</Fragment>
`;

exports[`Profile component renders the profile in a loading state 1`] = `
<Fragment>
  <Loading
    loading={true}
  />
  <NavBar />
</Fragment>
`;
